creator-node/src/routes/nodeSync.js

```diff
@@ -273,6 +274,27 @@ async function _nodesync (req, walletPublicKeys, creatorNodeEndpoint) {
         // but tracks cannot be created until metadata and cover art files have been created.
         const trackFiles = fetchedCNodeUser.files.filter(file => file.trackUUID != null)
         const nonTrackFiles = fetchedCNodeUser.files.filter(file => file.trackUUID == null)
+
+        // Save all track files to disk in batches (to limit concurrent load)
+        for (let i = 0; i < trackFiles.length; i += TrackSaveConcurrencyLimit) {
+          const trackFilesSlice = trackFiles.slice(i, i + TrackSaveConcurrencyLimit)
+          req.logger.info(`TrackFiles saveFileForMultihash - processing trackFiles ${i} to ${i + TrackSaveConcurrencyLimit}...`)
+          await Promise.all(trackFilesSlice.map(
+            trackFile => saveFileForMultihash(req, trackFile.multihash, trackFile.storagePath)
+          ))
+        }
+        req.logger.info('Saved all track files to disk and ipfs.')
+
+        // Save all non-track files to disk in batches (to limit concurrent load)
+        for (let i = 0; i < nonTrackFiles.length; i += NonTrackFileSaveConcurrencyLimit) {
+          const nonTrackFilesSlice = nonTrackFiles.slice(i, i + NonTrackFileSaveConcurrencyLimit)
+          req.logger.info(`NonTrackFiles saveFileForMultihash - processing files ${i} to ${i + NonTrackFileSaveConcurrencyLimit}...`)
```

Comment:
does this handle directories correctly? how do we recreate the dir structure etc here

Reply:
yeah i think hareesh is right. we don't have anything right now which actually runs this codepath. i sampled some files on prod and the initially uploaded images (pre-creator) are never transferred from user metadata. the only images we have in directory form are those that a user uploads after they become a creator.

Reply:
Ran thru all of https://www.notion.so/audiusproject/Creator-Node-Sync-QA-Checklist-e6837df7c5b643a599f4d5156ad98d95

PTAL again.